internal/host/validator.go

```diff
@@ -417,271 +388,214 @@ func (v *validator) areTangServersReachable(c *validationContext) ValidationStat
+func (v *validator) isValidPlatformNetworkSettings(c *validationContext) (ValidationStatus, string) {
 	if c.inventory == nil {
-		return ValidationPending
+		return ValidationPending, "Missing inventory"
 	}
 	if c.inventory.SystemVendor == nil {
-		return ValidationError
+		return ValidationError, "Validation error"
 	}
 	if funk.ContainsString(invalidPlatforms, c.inventory.SystemVendor.ProductName) {
 		// In case there is no cluster validation is pending
 		if c.infraEnv != nil {
-			return ValidationSuccessSuppressOutput
+			return ValidationSuccessSuppressOutput, ""
 		} else {
 			//In case userManagedNetworking is true, we don't care about the platform
-			return boolValue(swag.BoolValue(c.cluster.UserManagedNetworking))
+			if boolValue(swag.BoolValue(c.cluster.UserManagedNetworking)) == ValidationSuccess {
+				return ValidationSuccess, fmt.Sprintf("Platform %s is allowed", c.inventory.SystemVendor.ProductName)
+			}
+			return ValidationFailure, fmt.Sprintf("Platform %s is allowed only for Single Node OpenShift or user-managed networking", c.inventory.SystemVendor.ProductName)
 		}
 	}
-	return ValidationSuccess
-}
-
-func (v *validator) printValidPlatformNetworkSettings(c *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		return fmt.Sprintf("Platform %s is allowed", c.inventory.SystemVendor.ProductName)
-	case ValidationFailure:
-		return fmt.Sprintf("Platform %s is allowed only for Single Node OpenShift or user-managed networking", c.inventory.SystemVendor.ProductName)
-	case ValidationPending:
-		return "Missing inventory"
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
-	}
+	return ValidationSuccess, fmt.Sprintf("Platform %s is allowed", c.inventory.SystemVendor.ProductName)
 }
 
-func (v *validator) printHasMemoryForRole(c *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		return fmt.Sprintf("Sufficient RAM for role %s", common.GetEffectiveRole(c.host))
-	case ValidationFailure:
-		return fmt.Sprintf("Require at least %s RAM for role %s, found only %s",
-			conversions.BytesToString(conversions.MibToBytes(c.clusterHostRequirements.Total.RAMMib)), common.GetEffectiveRole(c.host), conversions.BytesToString(c.inventory.Memory.PhysicalBytes))
-	case ValidationPending:
-		return "Missing inventory or role"
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
-	}
-}
-
-func (v *validator) belongsToMachineCidr(c *validationContext) ValidationStatus {
+func (v *validator) belongsToMachineCidr(c *validationContext) (ValidationStatus, string) {
 	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput
+		return ValidationSuccessSuppressOutput, ""
 	}
 	if swag.StringValue(c.cluster.Kind) == models.ClusterKindAddHostsCluster || (swag.BoolValue(c.cluster.UserManagedNetworking) && !common.IsSingleNodeCluster(c.cluster)) {
```

Comment:
Is there any reason for this condition now?
At the very least the `AddHostsCluster` case doesn't need to be inside this conditional.

The other one ends up being this I think. But it might be worth getting someone who actually knows this validation to sort out if this logic is correct.
```
if swag.BoolValue(c.cluster.UserManagedNetworking) && (swag.StringValue(c.cluster.Kind) == models.ClusterKindAddHostsCluster || !common.IsSingleNodeCluster(c.cluster)) {
```